kf_lib_data_ingest/validation/validation.py

```diff
@@ -0,0 +1,75 @@
+import sys
+import os
+import logging
+
+from kf_lib_data_ingest.common.io import read_df, write_json
+
+
+VALIDATION_OUTPUT_DIR = "validation_results"
+RESULTS_FILENAME = "validation_results.json"
+
+
+class Validator(object):
+    def __init__(self, output_dir=None, setup_logger=True):
+        """
+        Constructor
+        :param output_dir: Directory where validation files will be written
+        :type output_dir: str
+        :param setup_logger: Whether to setup a console logger. Set to True if
+        running standalone
+        :type setup_logger: bool
+        """
+        self.output_dir = output_dir or os.path.join(
+            os.getcwd(), VALIDATION_OUTPUT_DIR
+        )
+        os.makedirs(self.output_dir, exist_ok=True)
+
+        if setup_logger:
+            root = logging.getLogger()
+            root.setLevel(logging.INFO)
+            consoleHandler = logging.StreamHandler()
+            consoleHandler.setFormatter(
+                logging.Formatter(
+                    "%(asctime)s - %(name)s" " - %(levelname)s - %(message)s"
+                )
+            )
+            root.addHandler(logging.StreamHandler())
+        self.logger = logging.getLogger(type(self).__name__)
+
+    def validate(self, filepath_list, include_implicit=True, **report_kwargs):
+        """
+        Validate a set of tabular files containing a standardized set of
+        columns
+        Write validation report(s) (optionally in various formats) to disk
+        TODO
+        """
+        try:
+            # TODO
+            # File paths to dict of DataFrames
+            df_dict = {
```

Comment:
Can you change this bit to the following so that if there are any files that cannot be read in as a dataframe by `read_df` the code won't break:

```python
df_dict = {}
for file_path in filepath_list:
    try:
        df_dict[fn] = read_df(os.path.join(file_path, fn))
    except Exception:
        continue
```

Reply:
what's `fn`?

Reply:
oh oops, sorry that should be `file_path`